CMakeLists.txt

```diff
@@ -71,6 +76,11 @@ if (DOWNLOAD_GMP)
         URL https://gmplib.org/download/gmp/gmp-6.3.0.tar.xz
         INSTALL_DIR ${CMAKE_CURRENT_BINARY_DIR}/gmp
         CONFIGURE_COMMAND "<SOURCE_DIR>/configure" "--prefix=<INSTALL_DIR>" --disable-shared --enable-static
+        # The libgmp.a output is created by the install step but since GMP::GMP
+        # depends on the entire ExternalProject build, we can pretend it is
+        # created by the build.
+        # TODO: Once we depend on CMake 3.26, use INSTALL_BYPRODUCTS
+        BUILD_BYPRODUCTS "<INSTALL_DIR>/lib/libgmp.a"
```

Comment:
This change is unrelated and I can pull it into a separate commit. Just noticed it while building with many jobs.

Reply:
Does something break if you don't have this then?

Reply:
I got a build failure with DOWNLOAD_GMP and `ninja -j 64`. Should I submit it as a separate PR?

Reply:
Nah I was just trying to understand it. I still don't understand why it fails without this though. The docs just say:

> Files that will be generated by the build command but may or may not have their modification time updated by subsequent builds.